### DISCUSSION TITLE
Question about modeling data

### DISCUSSION THREAD

**Discussion by daxidngyn** [prisma/prisma#12443] [Q&A]
Hi there!

I'm currently trying to model some data, but unsure of how to approach this:

I have a **user model** and a **class model**, where a user can join multiple classes and classes can have multiple users.
In addition, within each class, the user has a specific role. I was thinking that I could create a new model, **student**, that contains a role field and has a relation with both the user and class model, but I feel that this is a little repetitive.

Is there a more effective way of how I would go about modeling this? Any help is appreciated!

What I have so far:
```
model User {
  id            String        @id @default(auto()) @map("_id") @db.ObjectId
  classes       Class[]       @relation(fields: [classIDs], references: [id])
  classIDs      String[]      @db.ObjectId
  classRoles    Student[]
}

model Class {
  id            String        @id @default(auto()) @map("_id") @db.ObjectId
  users         User[]        @relation(fields: [userIDs], references: [id])
  userIDs       String[]      @db.ObjectId
  students      Student[]
}

model Student {
  id            String        @id @default(auto()) @map("_id") @db.ObjectId
  user          User          @relation(fields: [userId], references: [id])
  userId        String        @db.ObjectId
  class         Class         @relation(fields: [classId], references: [id])
  classId       String        @db.ObjectId
  role          ClassRole     @default(STUDENT)
}
```

**Comment by nurul3101** [comment]
Hey @daxidngyn 👋 

I think this is effective and shouldn't have any issues.

Model `Student` should be needed every user could have different classes and every class could have different students.
It would make sense to have a connecting table between a user and a class that represents this relationship. I have seen this pattern being used many times.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
